src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -75,12 +75,17 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
 
                         var finder = new SymbolReferenceFinder(this, document, semanticModel, diagnostic, node, cancellationToken);
 
+                        // Caches so we don't produce the same data multiple times while searching 
+                        // all over the solution.
+                        var projectToAssembly = new ConcurrentDictionary<Project, AsyncLazy<IAssemblySymbol>>();
+                        var referenceToCompilation = new ConcurrentDictionary<PortableExecutableReference, Compilation>();
```

Comment:
@pharring used to complain about the size of `ConcurrentDictionary` and removed a bunch of our uses of them.  How many threads are we actually doing this search on, and how many entries do we expect in these dictionaries?